lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -1271,6 +1272,13 @@ namespace Js
 
         this->GetDebugContext()->GetProbeContainer()->Initialize(this);
 
+        isDebugContextInitialized = true;
+
+#if defined(_M_ARM32_OR_ARM64)
+        // We need to ensure that the above write to the isDebugContextInitialized is visible to the debugger thread.
+        MemoryBarrier();
```

Comment:
This was the outcome of a discussion with @digitalinfinity, the concern was that setting a Boolean value was not guaranteed to be committed to memory on ARM.